csharp/src/Apache.Arrow/Arrays/PrimitiveArrayBuilder.cs

```diff
@@ -141,15 +141,17 @@ public TBuilder Append(ReadOnlySpan<T> span)
         {
             int len = ValueBuffer.Length;
             ValueBuffer.Append(span);
-            ValidityBuffer.AppendRange(Enumerable.Repeat(true, ValueBuffer.Length - len));
+            int additionalBitsCount = ValueBuffer.Length - len;
+            ValidityBuffer.Reserve(additionalBitsCount).AppendRange(Enumerable.Repeat(true, additionalBitsCount));
```

Comment:
If you really wanted to optimize this, we should get rid of the `Enumerable.Repeat` as well.

Reply:
I think the suggestion is to add a new `AppendRange` overload/method to BitmapBuilder...

public BitmapBuilder AppendRange(bool value, int length) {
  // Set bits [Length:Length+length) to value
}

Reply:
@westonpace has the right idea above. Add a method to the ValidityBuffer that adds  the same `bool` value `length` times without allocating an `Enumerable.Repeat` object.

Reply:
Since this is existing code, we can fix this in a separate PR.

Reply:
Hello @eerhardt  and @westonpace, I added a required method and created new PR. Could you please take a look https://github.com/apache/arrow/pull/35342 ?